### PR TITLE
Crop and calibrate bindings

### PR DESCRIPTION
**Pull request by NoahRJohnson**
More bindings for the crop_and_calibrate task.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -69,9 +75,9 @@ from . import contrib\n\
...
-add_subdirectory("vgl" "vgl-build")
-add_subdirectory("vil" "vil-build")
-add_subdirectory("vnl" "vnl-build")
-add_subdirectory("vpgl" "vpgl-build")
-add_subdirectory("contrib" "contrib-build")
+add_subdirectory("vgl" "pyvxl_build/vgl-build")
+add_subdirectory("vil" "pyvxl_build/vil-build")
+add_subdirectory("vnl" "pyvxl_build/vnl-build")
+add_subdirectory("vpgl" "pyvxl_build/vpgl-build")
+add_subdirectory("contrib" "pyvxl_build/contrib-build")
```
Comment (andyneff) [inline CMakeLists.txt]: @NoahRJohnson Why did you add pyvxl_build here? In the [past](https://bitbucket.org/visionsystemsinc/core3d/src/c149d25a3195ca73ee4ec71b55536c24f1df2b77/Core3D/CMakeLists.txt#lines-30) we used `CMAKE_BINARY_DIR`, so 
```suggestion
add_subdirectory("contrib" "${CMAKE_BINARY_DIR}/pyvxl-contrib-build")
```

[Reference](https://www.johnlamp.net/cmake-tutorial-4-libraries-and-subdirectories.html#ToDoCore/unit_test/CMakeLists.txt-13)

File: contrib/CMakeLists.txt [hunk 2]
```diff
@@ -7,6 +7,10 @@ set(PYVXL_CONTRIB_MAKE_ALL FALSE CACHE BOOL "Build every optional vxl contrib mo
 # INDIVIDUAL GUARDS, inherit from ALL GUARD
 set(PYVXL_CONTRIB_MAKE_BPGL ${PYVXL_CONTRIB_MAKE_ALL} CACHE BOOL "Turn on BPGL build")
 set(PYVXL_CONTRIB_MAKE_BRAD ${PYVXL_CONTRIB_MAKE_ALL} CACHE BOOL "Turn on BRAD build")
+set(PYVXL_CONTRIB_MAKE_BRIP ${PYVXL_CONTRIB_MAKE_ALL} CACHE BOOL "Turn on BRIP build")
+set(PYVXL_CONTRIB_MAKE_BVGL ${PYVXL_CONTRIB_MAKE_ALL} CACHE BOOL "Turn on BVGL build")
```
Comment (andyneff) [inline contrib/CMakeLists.txt]: Is this used?

File: contrib/CMakeLists.txt [hunk 3]
```diff
@@ -32,3 +36,48 @@ if (PYVXL_CONTRIB_MAKE_BRAD)
...
+  install(DIRECTORY DESTINATION ${PYTHON_SITE}/vxl/contrib/brip)
+
+  # auto generate __init__ file
+  install(CODE "file(APPEND ${PYTHON_SITE}/vxl/contrib/__init__.py \"\nfrom . import brip\")")
+
+  # Recurse
+  add_subdirectory("brip" "brip-build")
+endif()
+
+# if (PYVXL_CONTRIB_MAKE_BVGL)
```
Comment (andyneff) [inline contrib/CMakeLists.txt]: If this is junk code, please remove

File: contrib/bpgl/algo/pybpgl_algo.cxx [hunk 4]
```diff
@@ -14,25 +14,19 @@
...
-{
-  m.def("heightmap_from_disparity_affine", &bpgl_heightmap_from_disparity<vpgl_affine_camera<double> >);
-
+namespace pyvxl {
+namespace bpgl {
+namespace algo {
+  void wrap_bpgl_algo(py::module &m)
+  {
+    m.def("heightmap_from_disparity_affine", &bpgl_heightmap_from_disparity<vpgl_affine_camera<double> >);
+  }}}
```
Comment (andyneff) [inline contrib/bpgl/algo/pybpgl_algo.cxx]: This should be

```suggestion
  }
}}}
```

File: contrib/bpgl/algo/pybpgl_algo.cxx [hunk 5]
```diff
@@ -14,25 +14,19 @@
...
 
-}}}
-
 PYBIND11_MODULE(_bpgl_algo, m)
 {
   m.doc() =  "Python bindings for the VXL computer vision libraries";
 
   pyvxl::bpgl::algo::wrap_bpgl_algo(m);
-
-/* #ifdef PYVXL_WITH_CONTRIB_BPGL */
```
Comment (andyneff) [inline contrib/bpgl/algo/pybpgl_algo.cxx]: If junk code, remove

File: contrib/brad/pybrad.cxx [hunk 6]
```diff
@@ -4,22 +4,106 @@
...
+  auto* reflectance_img = new vil_image_view<float>(ni, nj, np);
+  bool success = brad_estimate_reflectance_image(radiance_img, mdata, mean_reflectance, *reflectance_img, average_airlight, is_normalize);
+
+  if (!success)
+    throw std::runtime_error(std::string("ERROR: vxl.contrib.brad.estimate_reflectance: brad_estimate_reflectance_image failed.\n"));
+
+  return *reflectance_img;
+}
+
+/* std::tuple<unsigned int, > get_image_footprint() */
```
Comment (andyneff) [inline contrib/brad/pybrad.cxx]: If junk code, remove
Also C style comments?! What kind of editor are you USING!?

File: contrib/brip/__init__.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+from ._brip import *
+
+
+def truncate_nitf_image(img, is_byte, is_scale):
```
Comment (andyneff) [inline contrib/brip/__init__.py]: Does the user use this? Do _they_ have to know if it is byte or not before calling this?

File: contrib/brip/pybrip.h [hunk 8]
```diff
@@ -0,0 +1,12 @@
+#ifndef pybrip_h_included_
+#define pybrip_h_included_
+
+#include <pybind11/pybind11.h>
+
+namespace pyvxl { namespace brip {
+
+void wrap_brip(pybind11::module &m);
```
Comment (andyneff) [inline contrib/brip/pybrip.h]:
```suggestion
  void wrap_brip(pybind11::module &m);
```

File: contrib/bvxm/algo/__init__.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
+from ._bvxm_algo import *
+
+
+def create_scene(scene_xml):
```
Comment (andyneff) [inline contrib/bvxm/algo/__init__.py]: Woot!

File: contrib/bvxm/algo/pybvxm_algo.cxx [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+#include <bvxm/bvxm_voxel_world.h>
+#include <bvxm/bvxm_world_params.h>
+#include <bvxm/algo/bvxm_create_scene_xml.h>
+#include <vul/vul_file.h>
+
+namespace py = pybind11;
+
+namespace pyvxl { namespace bvxm { namespace algo {
+
+bvxm_voxel_world create_voxel_world(std::string const& vox_dir, std::string const& lvcs_path,
```
Comment (andyneff) [inline contrib/bvxm/algo/pybvxm_algo.cxx]: Indent

File: contrib/bvxm/algo/pybvxm_algo.h [hunk 11]
```diff
@@ -0,0 +1,12 @@
+#ifndef pybvxm_algo_h_included_
+#define pybvxm_algo_h_included_
+
+#include <pybind11/pybind11.h>
+
+namespace pyvxl { namespace bvxm { namespace algo {
+
+void wrap_bvxm_algo(pybind11::module &m);
```
Comment (andyneff) [inline contrib/bvxm/algo/pybvxm_algo.h]:
```suggestion
  void wrap_bvxm_algo(pybind11::module &m);
```

File: CMakeLists.txt (hunk 1 above)
Comment (NoahRJohnson) [reply CMakeLists.txt]: The idea was to congregate all of the different pyvxl submodule build paths under a single pyvxl build, so that the parent project's Build dir doesn't get cluttered. The paths are relative to the current output directory, but I can make them absolute by prepending ${CMAKE_BINARY_DIR} to them. This would leave us with paths like:
add_subdirectory("contrib" "${CMAKE_BINARY_DIR}/pyvxl_build/contrib-build")

File: contrib/CMakeLists.txt (hunk 2 above)
Comment (NoahRJohnson) [reply contrib/CMakeLists.txt]: BRIP is used. The BVGL bindings weren't needed so I never ended up committing them. I've removed the reference to BVGL.

File: contrib/CMakeLists.txt (hunk 3 above)
Comment (NoahRJohnson) [reply contrib/CMakeLists.txt]: Done, thanks for pointing that out.

File: contrib/bpgl/algo/pybpgl_algo.cxx (hunk 5 above)
Comment (NoahRJohnson) [reply contrib/bpgl/algo/pybpgl_algo.cxx]: ? This is already removing junk code.

File: contrib/brad/pybrad.cxx (hunk 6 above)
Comment (NoahRJohnson) [reply contrib/brad/pybrad.cxx]: Removed. I'm using the vim-commentary plugin :D

File: contrib/brip/__init__.py (hunk 7 above)
Comment (NoahRJohnson) [reply contrib/brip/__init__.py]: Yes, and yes. This can be changed in the future once we have native vil load working.

File: contrib/brip/pybrip.h (hunk 8 above)
Comment (NoahRJohnson) [reply contrib/brip/pybrip.h]: Should all the functions in the namespace be indented in the .cxx file as well?

File: contrib/bvxm/algo/pybvxm_algo.cxx (hunk 10 above)
Comment (NoahRJohnson) [reply contrib/bvxm/algo/pybvxm_algo.cxx]: I'm of the opinion that namespaces shouldn't be indented. It unnecessarily nests all of your code. If I remove all the indentation from the codebase to make it consistent would that be acceptable?
